### ISSUE TITLE
Query parameters doesn't work for "path" condition in queries from system.zookeeper

### ISSUE THREAD

**Issue by Alex-Burmak** [ClickHouse/ClickHouse#12561]
The following queries demonstrate the issue:
```
# clickhouse-client -q "select name from system.zookeeper where path = '/'"
clickhouse


# clickhouse-client -q "select name from system.zookeeper where path = {path:String}" --param_path "/"
Received exception from server (version 20.5.2):
Code: 36. DB::Exception: Received from localhost:9001. DB::Exception: SELECT from system.zookeeper table must contain condition like path = 'path' in WHERE clause..


# clickhouse-client --send_logs_level error  -q "select name from system.zookeeper where path = {path:String}" --param_path "/"
[sas-vrxctd96rqw5lw25.db.yandex.net] 2020.07.17 10:54:51.941425 [ 36945 ] {782590e0-9ade-4b94-9a0b-3d47fcf589cf} <Error> executeQuery: Code: 36, e.displayText() = DB::Exception: SELECT from system.zookeeper table must contain condition like path = 'path' in WHERE clause. (version 20.5.2.7 (official build)) (from [::1]:49574) (in query: SELECT name FROM system.zookeeper WHERE path = CAST('/', 'String')), Stack trace (when copying this message, always include the lines below):

0. /build/obj-x86_64-linux-gnu/../contrib/poco/Foundation/src/Exception.cpp:27: Poco::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ 0x10ed0da0 in /usr/lib/debug/usr/bin/clickhouse
1. /build/obj-x86_64-linux-gnu/../src/Common/Exception.cpp:37: DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ 0x95c923d in /usr/lib/debug/usr/bin/clickhouse
2. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/string:2134: DB::StorageSystemZooKeeper::fillData(std::__1::vector<COW<DB::IColumn>::mutable_ptr<DB::IColumn>, std::__1::allocator<COW<DB::IColumn>::mutable_ptr<DB::IColumn> > >&, DB::Context const&, DB::SelectQueryInfo const&) const (.cold) @ 0xd7604b2 in /usr/lib/debug/usr/bin/clickhouse
3. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/vector:656: DB::IStorageSystemOneBlock<DB::StorageSystemZooKeeper>::read(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, DB::SelectQueryInfo const&, DB::Context const&, DB::QueryProcessingStage::Enum, unsigned long, unsigned int) @ 0xd66fb69 in /usr/lib/debug/usr/bin/clickhouse
4. /build/obj-x86_64-linux-gnu/../src/Processors/QueryPlan/ReadFromStorageStep.cpp:38: DB::ReadFromStorageStep::ReadFromStorageStep(DB::TableStructureReadLockHolder, DB::SelectQueryOptions, std::__1::shared_ptr<DB::IStorage>, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, DB::SelectQueryInfo const&, std::__1::shared_ptr<DB::Context>, DB::QueryProcessingStage::Enum, unsigned long, unsigned long) @ 0xe9c449d in /usr/lib/debug/usr/bin/clickhouse
5. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/memory:3028: DB::InterpreterSelectQuery::executeFetchColumns(DB::QueryProcessingStage::Enum, DB::QueryPlan&, std::__1::shared_ptr<DB::PrewhereInfo> const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) @ 0xdd9c7cf in /usr/lib/debug/usr/bin/clickhouse
6. /build/obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectQuery.cpp:785: DB::InterpreterSelectQuery::executeImpl(DB::QueryPlan&, std::__1::shared_ptr<DB::IBlockInputStream> const&, std::__1::optional<DB::Pipe>) @ 0xdda0622 in /usr/lib/debug/usr/bin/clickhouse
7. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/optional:224: DB::InterpreterSelectQuery::buildQueryPlan(DB::QueryPlan&) @ 0xdda1bc4 in /usr/lib/debug/usr/bin/clickhouse
8. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/vector:1549: DB::InterpreterSelectWithUnionQuery::buildQueryPlan(DB::QueryPlan&) @ 0xdf04084 in /usr/lib/debug/usr/bin/clickhouse
9. /build/obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:200: DB::InterpreterSelectWithUnionQuery::execute() @ 0xdf043dc in /usr/lib/debug/usr/bin/clickhouse
10. /build/obj-x86_64-linux-gnu/../src/Interpreters/executeQuery.cpp:384: DB::executeQueryImpl(char const*, char const*, DB::Context&, bool, DB::QueryProcessingStage::Enum, bool, DB::ReadBuffer*) @ 0xe074a59 in /usr/lib/debug/usr/bin/clickhouse
11. /build/obj-x86_64-linux-gnu/../src/Interpreters/executeQuery.cpp:643: DB::executeQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::Context&, bool, DB::QueryProcessingStage::Enum, bool) @ 0xe07811a in /usr/lib/debug/usr/bin/clickhouse
12. /build/obj-x86_64-linux-gnu/../src/Server/TCPHandler.cpp:251: DB::TCPHandler::runImpl() @ 0xe698946 in /usr/lib/debug/usr/bin/clickhouse
13. /build/obj-x86_64-linux-gnu/../src/Server/TCPHandler.cpp:1197: DB::TCPHandler::run() @ 0xe699660 in /usr/lib/debug/usr/bin/clickhouse
14. /build/obj-x86_64-linux-gnu/../contrib/poco/Net/src/TCPServerConnection.cpp:57: Poco::Net::TCPServerConnection::start() @ 0x10deebcb in /usr/lib/debug/usr/bin/clickhouse
15. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/atomic:856: Poco::Net::TCPServerDispatcher::run() @ 0x10def05b in /usr/lib/debug/usr/bin/clickhouse
16. /build/obj-x86_64-linux-gnu/../contrib/poco/Foundation/include/Poco/Mutex_POSIX.h:59: Poco::PooledThread::run() @ 0x10f6db86 in /usr/lib/debug/usr/bin/clickhouse
17. /build/obj-x86_64-linux-gnu/../contrib/poco/Foundation/include/Poco/AutoPtr.h:223: Poco::ThreadImpl::runnableEntry(void*) @ 0x10f68f80 in /usr/lib/debug/usr/bin/clickhouse
18. start_thread @ 0x76db in /lib/x86_64-linux-gnu/libpthread-2.27.so
19. clone @ 0x121a3f in /lib/x86_64-linux-gnu/libc-2.27.so

Received exception from server (version 20.5.2):
Code: 36. DB::Exception: Received from localhost:9001. DB::Exception: SELECT from system.zookeeper table must contain condition like path = 'path' in WHERE clause..
```

The issue is that the parameter substitutes with `CAST('/', 'String'))` which is not handled by `system.zookeeper` table.

The related source code:
https://github.com/ClickHouse/ClickHouse/blob/0c903d3fdc802973cbf0a2b2c80db5df8bae8f91/src/Storages/System/StorageSystemZooKeeper.cpp#L45

**Comment by alexey-milovidov** [comment]
Can be implemented similar to `StorageDistributed::skipUnusedShards` or `transformQueryForExternalDatabase`.

**Comment by alexey-milovidov** [comment]
Or `evaluateConstantExpressionAsLiteral`.

**Comment by tavplubix** [comment]
Fixed in 91cbd49e85ee3de6c93719e17bbf7e7e422d889e
Test will be added in #21661
